packages/starlight/__tests__/test-config.ts

```diff
@@ -18,14 +21,29 @@ export function defineVitestConfig(
 	const build = opts?.build ?? { format: 'directory' };
 	const trailingSlash = opts?.trailingSlash ?? 'ignore';
 
+	const { starlightConfig } = await runPlugins(config, plugins, createTestPluginContext());
 	return getViteConfig({
 		plugins: [
-			vitePluginStarlightUserConfig(StarlightConfigSchema.parse(config), {
-				root,
-				srcDir,
-				build,
-				trailingSlash,
-			}),
+			vitePluginStarlightUserConfig(starlightConfig, { root, srcDir, build, trailingSlash }),
 		],
 	});
 }
+
+export function createTestPluginContext(): StarlightPluginContext {
+	return {
+		command: 'dev',
+		config: {} as StarlightPluginContext['config'],
+		isRestart: false,
+		logger: new TestAstroIntegrationLogger(),
+	};
+}
+
+class TestAstroIntegrationLogger {
+	options = {} as AstroIntegrationLogger['options'];
+	constructor(public label = 'test-integration-logger') {}
+	fork = (label: string) => new TestAstroIntegrationLogger(label);
+	info = () => undefined;
+	warn = () => undefined;
+	error = () => undefined;
+	debug = () => undefined;
+}
```

Comment:
This was actually where I started writing this code but unfortunately it seems to not be possible as `test-utils` cannot be imported in `test-config` because it imports `vitest` and would give the following error:

```
Error: Vitest failed to access its internal state.
```

Reply:
Ahhhh, got it. Wondered if there might be something like that. OK then I guess we can leave it like this for now. Other option would be a separate file like `test-plugin-utils` or something, but probably overkill for so little code.